android/src/main/java/com/swmansion/rnscreens/ScreenFragment.kt

```diff
@@ -25,15 +25,18 @@ import com.swmansion.rnscreens.events.ScreenWillDisappearEvent
 import kotlin.math.max
 import kotlin.math.min
 
-open class ScreenFragment : Fragment {
+open class ScreenFragment : Fragment, ScreenFragmentWrapper {
     enum class ScreenLifecycleEvent {
         Appear, WillAppear, Disappear, WillDisappear
     }
 
+    override val fragment: Fragment
+        get() = this
+
```

Comment:
Fragment subclass itself implements the new interface. 

Thanks to `fragment` property:

```kotlin
override val fragment: Fragment
    get() = this
```

we can refer to this specific `Fragment` instance inside `ScreenContainer`.

Please note also that we are not creating any strong-reference cycle here as this property is computed on-the-fly.